internal/image/ova.go

```diff
@@ -30,7 +30,6 @@ type Image interface {
 
 func (o *Ova) Generate() error {
 	tw := tar.NewWriter(o.Writer)
-	defer tw.Close()
```

Comment:
Why are you removing the call to `tw.Close`?
```
if err := tw.Close(); err != nil {
  // log.Fatal(err) or handlet it somehow...
}
```

Reply:
Also, it might be better to perform I/O ops off the main thread...

Reply:
> Why are you removing the call to `tw.Close`?
> 
> ```
> if err := tw.Close(); err != nil {
>   // log.Fatal(err) or handlet it somehow...
> }
> ```

Actually that was a reason why it previously didn't return 400, but instead returned 200. The HTTP stream was closed after this method was called, but actually after this method the openapi is checking the return object of the `GetSourceImage` method and set the specific return code, but since the stream was already closed it couldn't write proper return code.

Reply:
> Also, it might be better to perform I/O ops off the main thread...

Good point. I will create a task for it. And improve later.